asciidoctorj-core/src/main/java/org/asciidoctor/ast/Document.java

```diff
@@ -58,4 +58,20 @@
      * @return options defined in document.
      */
     Map<Object, Object> getOptions();
+
+    /**
+     * Gets the current counter with the given name and increases its value.
+     * @param name
+     * @return
+     */
+    int getCounter(String name);

```

Comment:
I think these methods should be named getAndIncrementCounter so we clearly communicate that this is a mutable operation.